tests/phpunit/tests/test-static-pages.php

```diff
@@ -312,6 +312,42 @@ function test_page_for_posts() {
 		$this->assertEquals( array( get_post( $en ) ), $GLOBALS['wp_query']->posts );
 	}
 
+	function test_untranslated_page_for_posts_2() {
+		wp_delete_post( self::$posts_fr, true );
+
+		$en = $this->factory->post->create(
+			array(
+				'post_title' => 'english post',
+				'post_type'  => 'post',
+			)
+		);
+		self::$polylang->model->post->set_language( $en, 'en' );
+
+		$page_en = $en = $this->factory->post->create(
+			array(
+				'post_title' => 'page en',
+				'post_type'  => 'page',
+			)
+		);
+		self::$polylang->model->post->set_language( $en, 'en' );
+		$page_fr = $fr = $this->factory->post->create(
+			array(
+				'post_title' => 'page fr',
+				'post_type'  => 'page',
+			)
+		);
+		self::$polylang->model->post->set_language( $fr, 'fr' );
+		self::$polylang->model->post->save_translations( $en, compact( 'en', 'fr' ) );
+
+		wp_delete_post( $page_fr, true );
```

Comment:
The languages are set in the cache as soon as in the [wpSetUpBeforeClass()](https://github.com/polylang/polylang/blob/f58b73b8ac7f5b621c9feb516083b0629668a5e8/tests/phpunit/tests/test-static-pages.php#L27) uses them.

Here, it doesn't seem that the `wp_delete_post()` call impacted the cached English language in the same manner than in production. You might check if some filters / actions are missing or should not have been set.